File: LeetCode/326. Power of Three/elixir/solution.exs
defmodule Solution do
  @spec is_power_of_three(n :: integer) :: boolean
  def is_power_of_three(n) when n > 1 do
    if rem(n, 3) == 0 do
      is_power_of_three(trunc(n / 3))
    else
      false
    end
  end

  def is_power_of_three(1) do
    true
  end

  def is_power_of_three(_) do
    false
  end
end

File: LeetCode/326. Power of Three/java/Solution.java
class Solution {
    public boolean isPowerOfThree(int n) {
        if (n == 1) {
            return true;
        }
        while (n > 3) {
            if (n % 3 != 0) {
                return false;
            }
            n /= 3;
        }
        return n == 3;
    }
}